### DISCUSSION TITLE
Union of a model with a Color raises error in creating `openAPI.json`.

### DISCUSSION THREAD

**Discussion by jorgecarleitao** [fastapi/fastapi#8160] [Questions] [answered]
**To Reproduce**

1. Create the following app (`main.py`):

```python
from typing import Union

from fastapi import FastAPI
from pydantic import BaseModel
from pydantic.color import Color


class A(BaseModel):
    type: str


class B(BaseModel):
    bla: Color


C = Union[A, B]

app = FastAPI()


@app.get('/{var}', response_model=C)
def get(var: int):
    if var == 1:
        return {'type': 'a'}
    else:
        return {'bla': '#ff00ff'}
```

2. run `uvicorn main:app`

3. open [http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs)

4. See error (`Failed to load API definition` on the browser), mini stack trace:

```
  File ".../venv/lib/python3.7/site-packages/pydantic/schema.py", line 772, in field_singleton_schema
    raise ValueError(f'Value not declarable with JSON Schema, field: {field}')
ValueError: Value not declarable with JSON Schema, field: bla type=Color required
```

5. [http://127.0.0.1:8000/1](http://127.0.0.1:8000/1) and [http://127.0.0.1:8000/2](http://127.0.0.1:8000/2), no error in shown.

**Expected behavior**
No error

**Environment:**
 - OS: [macOS]
 - FastAPI Version 0.31.0.
- Python version: python 3.7.3

**Additional context**
No additional context, plain FastAPI. I am not sure if this is coming from pydantic or fastAPI: it may be a pydantic problem, or how fastAPI is using pydantic.

**Comment by varunchopra** [comment]
Not sure if you're allowed to use `var` as it's a keyword.

**Comment by euri10** [comment]
it's not a bug in FastAPI @jorgecarleitao 
the reason it crashes however is that the quite new `Color` type has not been made available through the `.schema()` method which FastAPI uses under the hood from pydantic.

**Comment by jorgecarleitao** [comment]
@euri10, got it. Thanks. Is there anything to report to pydantic then? If yes, what would be the minimal example for them?

**Comment by euri10** [comment]
I just created a PR @jorgecarleitao

**Comment by jorgecarleitao** [comment]
Perfect. Thanks a lot, @euri10. I am closing this so that we monitor it from there.

**Comment by tiangolo** [comment, marked as answer]
Thanks @euri10 for your help here and implementing it in Pydantic! :cake: :rocket: 

Thanks @jorgecarleitao for reporting back and closing the issue.

Just in case, this should be available in a recent version of FastAPI.
